src/main/java/com/uber/cadence/internal/sync/DeterministicRunnerImpl.java

```diff
@@ -299,7 +300,10 @@ public void close() {
     }
     try {
       for (WorkflowThread c : threads) {
-        c.stop();
+        Future<?> future = c.stopNow();
+        if (future != null) {
```

Comment:
I would make this method always return a future. If it is already stopped it can return a future that is already ready.

Reply:
Done